sources/packages/api/src/route-controllers/user/user.controller.ts

```diff
@@ -1,20 +1,49 @@
-import { Controller, Get, Req } from "@nestjs/common";
+import { Controller, Get, Patch, Req } from "@nestjs/common";
 import { Request } from "express";
 import { AuthService, UserService } from "../../services";
+import BaseController from "../BaseController";
 
 @Controller("users")
-export class UserController {
+export class UserController extends BaseController {
   constructor(
     private readonly service: UserService,
     private readonly authService: AuthService,
-  ) {}
+  ) {
+    super();
+  }
 
   @Get("/check-user")
-  checkUser(@Req() request: Request) {
-    const userInfo = this.authService.parseAuthorizationHeader(
-      request.headers.authorization,
-    );
+  async checkUser(@Req() request: Request) {
+    try {
+      const userInfo = this.authService.parseAuthorizationHeader(
+        request.headers.authorization,
+      );
+      const userInSABC = await this.service.getUser(userInfo.userName);
+      if (!userInSABC) {
+        //TODO: Set a 404 in response when no user is found.
+        //response.status(404);
+      } else {
+        return userInSABC;
+      }
+    } catch (error) {
+      this.handleRequestError(error);
+      throw error;
+    }
+  }
 
-    return this.service.getUser(userInfo.userName);
+  @Patch("/sync-user")
+  async synchronizeUserInfo(@Req() request: Request) {
+    try {
+      const userInfoBCServiceCard = this.authService.parseAuthorizationHeader(
+        request.headers.authorization,
+      );
+      const syncedUser = await this.service.synchronizeUserInfo(
+        userInfoBCServiceCard,
+      );
+      return syncedUser;
```

Comment:
Is this the object from the database? In general, I believe that we should avoid returning objects direct from the database model.
My understanding is that we should create models to return the exact information needed for every single information being returned form the API and being submitted.
@fwpushan, is it of for you also?